openvdb/points/PointMove.h

```diff
@@ -0,0 +1,997 @@
+                targetHandle.set(targetOffset, i, sourceValue);
+            }
+        }
+    }
+
+    void performMove(int targetOffset, int sourceOffset, const LeafT& targetLeaf,
+        IndexArray& offsets, const IndexPairArray& indices) const
+    {
+        auto& targetArray = mTargetHandles.getArray(targetOffset);
+        const auto& sourceArray = mSourceHandles.getConstArray(sourceOffset);
+
+        for (const auto& it : indices) {
+            const Index targetOffset = indexOffsetFromVoxel(it.first, targetLeaf, offsets);
+            targetArray.set(targetOffset, sourceArray, it.second);
+        }
+    }
+
+    void operator()(const LeafT& leaf, size_t idx) const
+    {
+        const auto& moveIndices = mMoveLeafMap.at(idx);
+        if (moveIndices.empty())  return;
+
+        // extract target leaf and per-voxel offsets for this leaf
+
+        auto& offsets = mOffsetMap[idx];
+
+        // extract source leaf that has the same origin as the target leaf (if any)
+
+        const auto it = mTargetToSourceMap.find(idx);
+        assert(it != mTargetToSourceMap.end());
+
+        const Index sourceOffset(it->second);
+
+        const auto& array = leaf.constAttributeArray(mAttributeIndex);
+
+        PerformTypedMoveOp op(mTargetHandles, mSourceHandles,
+            idx, sourceOffset, leaf, offsets, moveIndices);
+        if (!processTypedArray(array, op)) {
+            this->performMove(idx, sourceOffset, leaf, offsets, moveIndices);
+        }
+    }
+
+private:
+    OffsetMap& mOffsetMap;
+    AttributeHandles& mTargetHandles;
+    const IndexToIndexMap& mTargetToSourceMap;
+    AttributeHandles& mSourceHandles;
+    const Index mAttributeIndex;
+    const LocalPointIndexMap& mMoveLeafMap;
+}; // struct LocalMovePointsOp
+
+} // namespace point_move_internal
+
+
+////////////////////////////////////////
+
+
+template <typename PointDataGridT, typename DeformerT, typename FilterT>
+inline void movePoints( PointDataGridT& points,
+                        const math::Transform& transform,
+                        DeformerT& deformer,
+                        const FilterT& filter,
+                        future::Advect* objectNotInUse,
+                        bool threaded)
+{
+    using PointDataTreeT = typename PointDataGridT::TreeType;
+    using LeafT = typename PointDataTreeT::LeafNodeType;
+    using LeafManagerT = typename tree::LeafManager<PointDataTreeT>;
+    using LeafRangeT = typename LeafManagerT::LeafRange;
+
+    using namespace point_move_internal;
+
+    // this object is for future use only
+    assert(!objectNotInUse);
+
+    PointDataTreeT& tree = points.tree();
+
+    // early exit if no LeafNodes
+
+    PointDataTree::LeafCIter iter = tree.cbeginLeaf();
+
+    if (!iter)      return;
+
+    // build voxel topology taking into account any point group deletion
+
+    auto newPoints = point_mask_internal::convertPointsToScalar<PointDataGrid>(
+        points, transform, filter, deformer, threaded);
```

Comment:
While it's true that the pdg being generated isn't valid (yet), there's nothing in the API to prevent partial creation of a pdg and I feel it should be encouraged in cases like there where there's some efficiency to be gained in doing so. After all, this is an "internal" method calling another "internal" method in a way that isn't explicitly documented as part of the public API.